kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/sink/seek/IndexManager.scala

```diff
@@ -0,0 +1,184 @@
+          } else {
+            storageInterface
+              .deleteFiles(mostRecentIndexFile.bucket, filtered)
+              .map(_ => {
+                logger.debug("[{}] Retaining index file: {}, Deleting files: ({})", sinkName, mostRecentIndexFile.path, filtered.mkString(", "))
+                filtered.size
+              })
+              .leftMap {
+                err =>
+                  logger.error("[{}] Error on cleanup: {}", sinkName, err.message(), err.exception)
+              }.getOrElse(0).asRight
+          }
+      }
+  }
+
+  private def logAndReturnMaxExceededError(topicPartition: TopicPartition, indexes: List[String]) = {
+    val logLine = s"Too many index files have accumulated (${indexes.size} out of max $maxIndexes)"
+    logger.error(s"[{}] {}", sinkName, logLine)
+    FatalS3SinkError(logLine, topicPartition).asLeft
+  }
+
+  def write(topicPartitionOffset: TopicPartitionOffset, s3PathLocation: RemoteS3PathLocation): Either[SinkError, RemoteS3PathLocation] = {
+
+    val s3Path = IndexFilenames.indexFilename(
+      sinkName,
+      topicPartitionOffset.topic.value,
+      topicPartitionOffset.partition,
+      topicPartitionOffset.offset.value
+    )
+
+    logger.debug("[{}] Writing index {} pointing to file {}", sinkName, s3Path, s3PathLocation.path)
+
+    val idxPath = s3PathLocation.root().withPath(s3Path)
+    storageInterface
+      .writeStringToFile(idxPath, s3PathLocation.path)
+      .map(_ => idxPath)
+      .leftMap(
+        ex => {
+          logger.error("[{}] Exception writing index {} pointing to file {}", sinkName, s3Path, s3PathLocation.path, ex)
+          NonFatalS3SinkError(ex.message())
+        }
+      )
+
+  }
+
+  def fallbackOffsetSeek(seeker: LegacyOffsetSeeker, topicPartition: TopicPartition, fileNamingStrategy: S3FileNamingStrategy, bucketAndPrefix: RemoteS3RootLocation): Either[SinkError, Option[TopicPartitionOffset]] = {
+    seeker.seek(topicPartition, fileNamingStrategy, bucketAndPrefix) match {
+      case Left(err) =>
+        logger.error(s"[{}] Error retrieving fallback offset seek topicPartition {} for {}", sinkName, topicPartition, bucketAndPrefix.bucket)
+        err.asLeft
+      case Right(Some((tpo, path))) =>
+        write(tpo, path).map(_ => Some(tpo))
+      case Right(None) =>
+        None.asRight
+    }
+  }
+
+  /**
+   * Seeks the filesystem to find the latyest offsets for a topic/partition.
+   * @param topicPartition the TopicPartition for which to retrieve the offsets
+   * @param fileNamingStrategy the S3FileNamingStrategy to use in the case that a fallback offset seeker is required.
+   * @param bucketAndPrefix the configured RemoteS3RootLocation
+   * @return either a SinkError or an option to a TopicPartitionOffset with the seek result.
+   */
+  def seek(topicPartition: TopicPartition, fileNamingStrategy: S3FileNamingStrategy, bucketAndPrefix: RemoteS3RootLocation): Either[SinkError, Option[TopicPartitionOffset]] = {
+
+    /**
```

Comment:
why is this indentation different from above

Reply:
It's a nested function.

Reply:
ah, sorry about this.. maybe i need to look better/more and get new glasses. :)